src/integration/proxy.rs

```diff
@@ -0,0 +1,141 @@
+use bytes::Bytes;
+use http::header;
+use http_body_util::{combinators::BoxBody, BodyExt, Full};
+use hyper::client::conn::http1::Builder;
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Request, Response};
+use hyper_util::rt::TokioIo;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::net::{TcpListener, TcpStream};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct JsonRpcRequest {
+    jsonrpc: String,
+    method: String,
+    params: Value,
+    id: Value,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct JsonRpcResponse {
+    jsonrpc: String,
+    #[serde(default)]
+    result: Option<Value>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    error: Option<JsonRpcError>,
+    id: Value,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct JsonRpcError {
+    code: i32,
+    message: String,
+    data: Option<Value>,
+}
```

Comment:
don't these types already exist in the rpc library?

Reply:
I tried to use them but they had some generics that I did not manage to figure out. Feel free to try.